pkg/reconcile/pipeline/handler/collect/impl_test.go

```diff
@@ -388,6 +388,62 @@ var _ = Describe("Collect Binding Data", func() {
 			shouldRetry(pipeline.HandlerFunc(collect.BindingItems), collect.ErrorReadingBindingReason, err)
 		})
 
+		Context("on invalid sourceValue in sliceOfMap elements", func() {
+			var (
+				mockCtrl *gomock.Controller
+			)
+
+			BeforeEach(func() {
+				mockCtrl = gomock.NewController(GinkgoT())
+			})
+
+			AfterEach(func() {
+				mockCtrl.Finish()
+			})
+
+			It("should set an error condition",
+				func() {
+					serviceContent := map[string]interface{}{
+						"metadata": map[string]interface{}{
+							"annotations": map[string]interface{}{
+								"service.binding/java-maven_port": "path={.status.ports},elementType=sliceOfMaps,sourceKey=name,sourceValue=asdf",
+							},
+						},
+						"status": map[string]interface{}{
+							"ports": []interface{}{
+								map[string]interface{}{"name": "foo", "value": "bar"},
+							},
+						},
+					}
+
+					ctx := mocks.NewMockContext(mockCtrl)
+					service := mocks.NewMockService(mockCtrl)
+					serviceResource := &unstructured.Unstructured{Object: serviceContent}
+					definition := bindingmocks.NewMockDefinition(mockCtrl)
+					value := bindingmocks.NewMockValue(mockCtrl)
+
+					ctx.EXPECT().Services().Return([]pipeline.Service{service}, nil)
+
+					bindingDefs := []binding.Definition{definition}
+					service.EXPECT().BindingDefs().DoAndReturn(func() []binding.Definition { return bindingDefs })
+					service.EXPECT().Resource().Return(serviceResource)
+
+					definition.EXPECT().Apply(serviceResource).Return(value, nil)
```

Comment:
given that we really mock this call, `serviceResource` value could be further simplified and reduce even to something like `unstructured.Unstructured{}`